CMakeLists.txt

```diff
@@ -95,9 +95,26 @@ include_directories(src)
 message (STATUS "Compiler ID: " ${CMAKE_CXX_COMPILER_ID})
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    add_compile_options("/we 4265") # Class has virtual functions, but its non-trivial destructor is not virtual
+    # /we #### - treat warning as error
+    # /wd #### - Disable warning
+    # /w3 #### - set warning to level 3
+    add_compile_options("/Wall")    # Enable all warnings
     add_compile_options("/w3 5038") # Data member will be initialized after [other] data member
+    add_compile_options("/we 4265") # Class has virtual functions, but its non-trivial destructor is not virtual
     add_compile_options("/we 5204") # Class has virtual functions, but its trivial destructor is not virtual
+    add_compile_options("/wd 4100") # unreferenced formal parameter
+    add_compile_options("/wd 4365") # conversion from 'size_t' to 'const int __int64', signed/unsigned mismatch
+    add_compile_options("/wd 4457") # index X range checked by comparison on this line
+    add_compile_options("/wd 4464") # feeds call on this line
```

Comment:
The MSVC docs refer to this as `relative include path contains '..'`: https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4464?view=msvc-170